protocols/v2/binary-sv2/no-serde-sv2/codec/src/codec/mod.rs

```diff
@@ -13,52 +67,97 @@ use buffer_sv2::Slice;
 
 use alloc::vec::Vec;
 
-/// Return the encoded byte size or a `Decodable`
+/// The `SizeHint` trait provides a mechanism to return the encoded bytes size of a decodable type.
+///
+/// It defines two methods for retrieving the size of an encoded message:
+///
+/// - `size_hint` is a static method that takes the raw data and an offset and returns the total
+///   size of the encoded message. This is particularly useful for types where the encoded size may
+///   vary based on the contents of the data, and we need to calculate how much space is required
+///   for decoding.
+/// - `size_hint_` is an instance method that performs the same function but allows the size to be
+///   be determined with respect to the current instance of the type.
```

Comment:
```suggestion
```
This is a duplicate.

Also this is a place to explain how to implement/use the trait in general and not specific for the functions.